### ISSUE TITLE
Implement new error structure in backend.

### ISSUE THREAD

**Issue by kgodey** [centerofci/mathesar#883]
## Problem

We've agreed on a standard error structure for the API in #560. Now we need to implement it.

## Proposed solution

I'll format these in terms of desired outcomes. Whoever works on this can feel free to split these up into separate issues if that's helpful.
- All current API error responses should follow our error structure.
- All error codes and their corresponding `details` fields are documented somewhere.
- We have a standard way of generating new error codes.

## Additional context

- #560 
- [API Standards wiki page](https://wiki.mathesar.org/en/engineering/architecture/api-standards)

**Comment by dmos62** [comment]
This seems up my valley. If @silentninja and @mathemancer don't mind, I'm taking this.

**Comment by silentninja** [comment]
I am planning to set up the necessary **infrastructure** so that it is easier to capture and define the error structure for the captured error. Which it be then followed by  each error getting then own error structure(sub-task list or individual issue), where #859 will be the priority

**Comment by kgodey** [comment]
@silentninja That's fine. please make small and incremental PRs so that changes are easy to review and merge. Getting #859 done to unblock the frontend soon should be a priority, so please organize your work so that we get to that ASAP.

**Comment by kgodey** [comment]
Thanks @silentninja
